pkg/systemd/kernelopt.sh

```diff
@@ -29,9 +29,23 @@ grub() {
     if type grubby >/dev/null 2>&1; then
         if [ "$1" = set ]; then
             grubby --args="$2" --update-kernel=ALL
+            # HACK: grubby on RHEL 8.0 does not change default kernel args (https://bugzilla.redhat.com/show_bug.cgi?id=1690765)
+            envopts=$(grub2-editenv - list | grep ^kernelopts) || envopts=""
+            if [ -n "$envopts" ]; then
+                newenvopts=$(echo "$envopts" | sed -r "s/$key(=[^[:space:]\"]*)?/$2/g; t; s/$/ $2/")
+            fi
         else
             grubby --remove-args="$2" --update-kernel=ALL
+            envopts=$(grub2-editenv - list | grep ^kernelopts) || envopts=""
+            if [ -n "$envopts" ]; then
+                newenvopts=$(echo "$envopts" | sed -r "s/$key(=[^[:space:]\"]*)?//g")
+            fi
         fi
+
+        if [ -n "$envopts" ] && [ "$newenvopts" != "$envopts" ]; then
+            grub2-editenv - set "$newenvopts"
```

Comment:
Just to make sure I understand this correctly, RHEL8 does allow setting of kernel args, but only through the environment?

Reply:
It's a little more complicated: In *principle*, RHEL 8 uses BLS, i. e. the initial kernel in a fresh install uses `options $kernelopts` in BLS entries (/boot/loader/entries/*.conf) and `kernelopts` is defined in grubenv. In Fedora 30 (where BLS is fully implemented), `grubby --args=foo --update-kernel=ALL` adds `foo` to grubenv, and leaves all BLS entries alone. But in RHEL 8.0, that command "expands" `options` in the BLS entries with the new option, does *not* update grubenv, and subsequent kernel installs copy that from the running kernel -- i. e. future kernels will miss the option changes. You can try that on our fedora-30 and rhel-8-0 images. This was discussed in the referenced bugzilla.

So, RHEL 8 (as well as Fedora 30 and BLS in general) allows setting options through BLS entries and through grubenv's `$kernelopts`, but the tools don't work well together and produce a mix-and-match.

Reply:
Right, I tihnk i understand it, thanks for the explanation! :)